conduit/pipeline/pipeline.go

```diff
@@ -269,76 +359,80 @@ func (p *pipelineImpl) Init() error {
 			return fmt.Errorf("Pipeline.Init(): could not read metadata: %w", err)
 		}
 	}
-	// overriding NextRound if NextRoundOverride is set
-	if p.cfg.ConduitArgs.NextRoundOverride > 0 {
-		p.logger.Infof("Overriding default next round from %d to %d.", p.pipelineMetadata.NextRound, p.cfg.ConduitArgs.NextRoundOverride)
-		p.pipelineMetadata.NextRound = p.cfg.ConduitArgs.NextRoundOverride
+
+	// Check for round override
+	p.logger.Infof("Checking for round override")
+	pluginRoundOverride, err := p.pluginRoundOverride()
+	if err != nil {
+		return fmt.Errorf("Pipeline.Init(): error resolving plugin round override: %w", err)
 	}
+	if pluginRoundOverride > 0 && pluginRoundOverride != p.pipelineMetadata.NextRound {
+		p.logger.Infof("Overriding default next round from %d to %d.", p.pipelineMetadata.NextRound, pluginRoundOverride)
+		p.pipelineMetadata.NextRound = pluginRoundOverride
+	} else {
+		p.logger.Infof("Initializing to pipeline round %d.", p.pipelineMetadata.NextRound)
+	}
+
 	// InitProvider
 	round := sdk.Round(p.pipelineMetadata.NextRound)
 	// Initial genesis object is nil--gets updated after importer.Init
 	var initProvider data.InitProvider = conduit.MakePipelineInitProvider(&round, nil)
 	p.initProvider = &initProvider
 
-	configs, err := yaml.Marshal(p.cfg.Importer.Config)
-	if err != nil {
-		return fmt.Errorf("Pipeline.Init(): could not serialize Importer.Args: %w", err)
-	}
-	genesis, err := (*p.importer).Init(p.ctx, *p.initProvider, p.makeConfig("importer", importerName, configs), importerLogger)
-	if err != nil {
-		return fmt.Errorf("Pipeline.Init(): could not initialize importer (%s): %w", importerName, err)
-	}
-	(*p.initProvider).SetGenesis(genesis)
+	// Initialize Importer
+	{
```

Comment:
Why does this need a scope?

Reply:
It's not strictly needed. In this case I think it's nice to have all this code in one function, but the variables leak all over the place. Adding in a scope means that there aren't as many variables floating around.

I didn't want to completely refactor because that will come when we do the data pipe-lining work.